### DISCUSSION TITLE
Row action button with permalink URL

### DISCUSSION THREAD

**Discussion by allgood** [Power-Components/livewire-powergrid#662] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

Sample code: 

```php
        return [
           Button::add('go to page')
           ->caption('<i class="fas fa-edit"></i>')
           ->route('page',
               [
                   'parameter' => 'id'                   
               ] )
            ->target('_self')  
            ->tooltip('Go to page')
       ]
```
Row action button generates links using URL parameters like `https://myapp.example.com/page?parameter=value`, would be great if there was a possibility to generate URL with a permalink style like `https://myapp.example.com/page/value`

### Why is this needed?

Usual way to declare routes on laravel is like this:

```php
Route::get('/page/{parameter}', function () {
    return view('page');
})->name('page');
```
Currently when using action button to launch such pages there is a need to create another route without the parameter and read it from the URL parameter on the template with the `request()->get('parameter')` instead of the usual `request()->route('parameter')` 

### Suggested Solution(s)

This could be made using non-associative array, like the sample code bellow:

```php
        return [
           Button::add('go to page')
           ->caption('<i class="fas fa-edit"></i>')
           ->route('page',
               [
                   'id'                   
               ] )
            ->target('_self')  
            ->tooltip('Go to page')
       ]
```

**Comment by luanfreitasdev** [comment]
Hi, thank you very much for the idea. Can you submit a PR for this solution?. 

As there are no break changes you can enter V3.

Maybe we can rename it to redirect ?

**Comment by allgood** [comment]
After I wrote the feature request I noticed that `route()` method have a `singleParam` boolean parameter that isn't on the documentation. What is it for exactly?

**Comment by allgood** [comment]
Just noted that this is already supported... just need to point to a route that have the parameter defined.

tested with:

On `web.php`:
```php
Route::view('/powergrid/{id}', 'powergrid-demo')->name('powergrid-id');
```

and this when defining the actions:

```php
    public function actions(): array
    {
       return [
           Button::make('info', 'Click me')
               ->route('powergrid-id', ['id' => 'id']),
        ];
    }
```

This was sufficient to get the links generated correctly.

# CONCLUSION

**The feature already exists**

**Reply by allgood** [reply]
I made this test with the powergrid generated demo, will now try to check why it didn't worked on my code.
